rpython/flowspace/flowcontext.py

```diff
@@ -374,6 +381,28 @@ def maybe_merge(self):
 
     def record(self, spaceop):
         spaceop.offset = self.last_offset
+
+        # Add a comment giving the source-code location of this operation
+        # e.g. 'is_perfect_number:35 :     return n == sum'
+        try:
+            code = self.pycode
+            linenum = self.last_offset
+            src = self.graph.source.split('\n')[linenum]
+            text = '%s:%d : %s' % (code.co_name, linenum, src)
```

Comment:
This part fails. self.pycode is the compiled bytecode, and the `last_offset` relates to a bytecode offset, not a code linenum. Do we have any information about the line number in the source code anywhere?

Reply:
yes, you can call `offset2lineno(code, self.last_offset)` to get the line number. the function is in `rpython.tool.error`.